### PR TITLE
move v1alpha2 to networking/v1alpha3 crd instead of config

### PR DESCRIPTION
**Pull request by GregHanson**

### CODE REVIEW

**Comment by GregHanson** [comment]
related to discussion here:
https://github.com/istio/istio/issues/3906

does not fully fix, because crd client still needs to be able to handle apiversion v1alpha1 and v1alpha3 at same time

File: install/kubernetes/helm/istio/charts/pilot/templates/crds.yaml [hunk 1]
```diff
@@ -65,11 +65,11 @@ spec:
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
-  name: destinationrules.config.istio.io
+  name: destinationrules.networking.istio.io
```
Comment (xiaolanz) [inline install/kubernetes/helm/istio/charts/pilot/templates/crds.yaml]: need change version v1alpha2 -> v1alpha3 to align with proto message package path.

**Comment by ldemailly** [comment]
just curious, it wasn't possible to do both changes in 1 ? (why first alpha2->alpha3 and then package change?)

File: pilot/pkg/model/config.go [hunk 2]
```diff
@@ -300,6 +300,10 @@ const (
 
 	// NamespaceAll is a designated symbol for listing across all namespaces
 	NamespaceAll = ""
+
+	configGroup = "config"
```
Comment (xiaolanz) [inline pilot/pkg/model/config.go]: do not define constants here. it doesn't help to clarify the model. Please write "networking" or "config" in each proto schema. 

i plan to remove istioAPIVersion once the everything is done.

File: tests/e2e/tests/pilot/pilot_test.go [hunk 3]
```diff
@@ -68,8 +68,8 @@ func init() {
 		fmt.Sprintf("Auth mode for the tests (Choose from %s, %s, %s)", authModeEnable, authModeDisable, authModeBoth))
 	flag.BoolVar(&config.Mixer, "mixer", config.Mixer, "Enable / disable mixer.")
 	flag.BoolVar(&config.V1alpha1, "v1alpha1", config.V1alpha1, "Enable / disable v1alpha1 routing rules.")
-	flag.BoolVar(&config.V1alpha2, "v1alpha2", config.V1alpha2, "Enable / disable v1alpha2 routing rules.")
-	flag.BoolVar(&config.RDSv2, "rdsv2", false, "Enable RDSv2 for v1alpha2")
+	flag.BoolVar(&config.V1alpha3, "v1alpha3", config.V1alpha3, "Enable / disable v1alpha3 routing rules.")
+	flag.BoolVar(&config.RDSv3, "rdsv3", false, "Enable RDSv3 for v1alpha3")
```
Comment (ijsnellf) [inline tests/e2e/tests/pilot/pilot_test.go]: RDSv2 != v1alpha2/3. I think the v2 refers to the using the Envoy v2 API, not to v1alpha2/3. This should stay `RDSv2` in all references.

File: tests/e2e/tests/pilot/util/config.go [hunk 4]
```diff
@@ -53,8 +53,8 @@ type Config struct {
 	DebugImagesAndMode    bool
 	UseAutomaticInjection bool
 	V1alpha1              bool
-	V1alpha2              bool
-	RDSv2                 bool
+	V1alpha3              bool
+	RDSv3                 bool
```
Comment (ijsnellf) [inline tests/e2e/tests/pilot/util/config.go]: see above

File: tests/e2e/tests/pilot/util/environment.go [hunk 5]
```diff
@@ -112,7 +112,7 @@ type TemplateData struct {
 	Ingress                bool
 	Zipkin                 bool
 	UseAdmissionWebhook    bool
-	RDSv2                  bool
+	RDSv3                  bool
```
Comment (ijsnellf) [inline tests/e2e/tests/pilot/util/environment.go]: see above

File: tests/e2e/tests/pilot/util/environment.go [hunk 6]
```diff
@@ -181,7 +181,7 @@ func (e *Environment) ToTemplateData() TemplateData {
 		PilotCustomConfigFile:  e.PilotCustomConfigFile,
 		MixerCustomConfigFile:  e.MixerCustomConfigFile,
 		CABundle:               e.CABundle,
-		RDSv2:                  e.Config.RDSv2,
+		RDSv3:                  e.Config.RDSv3,
```
Comment (ijsnellf) [inline tests/e2e/tests/pilot/util/environment.go]: see above

**Comment by GregHanson** [comment]
@xiaolanz PTAL at most recent commits for requested changes

File: tests/e2e/tests/pilot/pilot_test.go (hunk 3 above)
Comment (ldemailly) [reply tests/e2e/tests/pilot/pilot_test.go]: did some test failed? (I would hope so but...)

**Comment by sebastienvas** [comment]
It seems that  istio-pilot-e2e has been consistently failing since this PR.

**Comment by douglas-reid** [comment]
Yes, as has e2e-bookInfoTests.

@GregHanson can you take a look ?

**Pull request merged**